### ISSUE TITLE
to_ratings_per_user function unclear 

### ISSUE THREAD

**Issue by franziweindel** [FlorianWilhelm/lda4rec#1]
Hi Florian, 

I am having troubles understanding what the function to_ratings_per_user does. Should it return the the number of total ratings each user has made or for each users the actual ratings?  
Also when running it I do get an index error since both of the keys in the df dataframe that is first defined are floats, so they cannot be used to index into the cube variable.

Any help is appreciated, thanks, 

Franzi

**Comment by FlorianWilhelm** [comment]
Hi @franziweindel, so basically the function constructs a matrix representing the mapping (u, i) -> v, where u is some user_id, i is the i-th interaction of the user with user_id u, and v is the item_id. The first dimension of the matrix is the number of users and the second dimension is the maximum number of interactions a user had. Since the matrix is rectangular, item_id -999 is used for padding. Does this explain it?

**Comment by franziweindel** [comment]
Hi @FlorianWilhelm, 

Thank you for your response and explaination! It is clear now! However, in my end results I now get negative entries for the topic-item distribution phi, does that have something to do with the padding or the negative sampling for the BPR loss function. 

Thank you, 

Franzi

**Comment by FlorianWilhelm** [comment]
Hi, it's the logits your are looking at. What are you trying to do btw?

**Comment by franziweindel** [comment]
ahh okay thanks a lot! 
I am writing my thesis about explainable RS with the goal of identifiying implicit preferences from consumer behavior and I am trying to implement your LDA model on my dataset right now , because I eventually want to extend it to a correlated LDA model with the additional bias and user conformity paramters : )

**Comment by FlorianWilhelm** [comment]
Nice! Let me know when your work is available, I am looking forward to read it.
